File: java-module/src/main/java/com/pfefferminzia/kfz/common/PlausiException.java
package com.pfefferminzia.kfz.common;

import java.util.List;

public class PlausiException extends RuntimeException {

    private final List<String> fehler;

    public PlausiException(List<String> fehler) {
        super(erstelleFehlermeldung(fehler));
        this.fehler = fehler;
        // Ausgabe in Console
        System.err.println(getMessage());
    }

    private static String erstelleFehlermeldung(List<String> fehler) {
        if (fehler == null || fehler.isEmpty()) {
            return "Plausibilitätsprüfung fehlgeschlagen";
        }

        StringBuilder sb = new StringBuilder();
        sb.append("\n");
        sb.append("=".repeat(70)).append("\n");
        sb.append("PLAUSIBILITÄTSFEHLER\n");
        sb.append("=".repeat(70)).append("\n");
        
        for (int i = 0; i < fehler.size(); i++) {
            sb.append(String.format("  %d. %s\n", i + 1, fehler.get(i)));
        }
        
        sb.append("=".repeat(70));
        
        return sb.toString();
    }

    public List<String> getFehler() {
        return fehler;
    }
}

File: dotnet-module/KfzVersicherung/Common/PlausiException.cs

namespace DotnetModule.Common;

public class PlausiException : Exception
{
    public List<string> Fehler { get; }

    public PlausiException(List<string> fehler) 
        : base(ErstelleFehlermeldung(fehler))
    {
        Fehler = fehler;
        // Ausgabe in Console
        Console.Error.WriteLine(Message);
    }

    private static string ErstelleFehlermeldung(List<string> fehler)
    {
        if (fehler.Count == 0)
        {
            return "Plausibilitätsprüfung fehlgeschlagen";
        }

        var sb = new System.Text.StringBuilder();
        sb.AppendLine();
        sb.AppendLine(new string('=', 70));
        sb.AppendLine("PLAUSIBILITÄTSFEHLER");
        sb.AppendLine(new string('=', 70));
        
        for (int i = 0; i < fehler.Count; i++)
        {
            sb.AppendLine($"  {i + 1}. {fehler[i]}");
        }
        
        sb.Append(new string('=', 70));
        
        return sb.ToString();
    }
}
